gcp_variant_transforms/transforms/bigquery_to_variant.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2018 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A PTransform to convert BigQuery table rows to a PCollection of `Variant`."""
+
+from typing import Any, Dict, List  # pylint: disable=unused-import
+
+import apache_beam as beam
+
+from gcp_variant_transforms.beam_io import vcfio
+from gcp_variant_transforms.libs import bigquery_util
+
+# Reserved constants for column names in the BigQuery schema.
+RESERVED_BQ_COLUMNS = [bigquery_util.ColumnKeyConstants.REFERENCE_NAME,
+                       bigquery_util.ColumnKeyConstants.START_POSITION,
+                       bigquery_util.ColumnKeyConstants.END_POSITION,
+                       bigquery_util.ColumnKeyConstants.REFERENCE_BASES,
+                       bigquery_util.ColumnKeyConstants.ALTERNATE_BASES,
+                       bigquery_util.ColumnKeyConstants.NAMES,
+                       bigquery_util.ColumnKeyConstants.QUALITY,
+                       bigquery_util.ColumnKeyConstants.FILTER,
+                       bigquery_util.ColumnKeyConstants.CALLS]
+
+RESERVED_VARIANT_CALL_COLUMNS = [
+    bigquery_util.ColumnKeyConstants.CALLS_NAME,
+    bigquery_util.ColumnKeyConstants.CALLS_GENOTYPE,
+    bigquery_util.ColumnKeyConstants.CALLS_PHASESET
+]
+
+
+class BigQueryToVariant(beam.PTransform):
+  """Transforms BigQuery table rows to PCollection of `Variant`."""
+
+  def expand(self, pcoll):
+    return (pcoll | 'BigQueryToVariant' >> beam.Map(
+        self._convert_bq_row_to_variant))
+
+  def _convert_bq_row_to_variant(self, row):
+    # type: (Dict[str, Any]) -> vcfio.Variant
+    return vcfio.Variant(
```

Comment:
I just noticed that this should be in a separate module (similar to VCF to BQ where the BQ row generation is inside /libs) so that it can be reused for other use cases. Also one of the PTransform 'styleguide'. Please just add a TODO for now.